### PR TITLE
[Feature] Support dynamic modification of datacache.partition_duration

### LINKED ISSUE #32840: [Feature] Support dynamic modification of datacache.partition_duration
## Feature request

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by stonechen111**
Modify the partition_duration parameter from read-only to dynamic modification. 
This facilitates users to modify table attributes in real time.

Fixes #32840

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [x] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

File: fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java [hunk 1]
```diff
@@ -1600,28 +1601,41 @@ public AlterJobV2 createAlterMetaJob(List<AlterClause> alterClauses, Database db
...
+                    throw new DdlException("Null datacache.partition_duration");
+                }
+
+                PeriodDuration oldPartitionDuration = olapTable.dataCachePartitionDuration();
+                if (partitionDuration.equals(oldPartitionDuration)) {
+                    LOG.info(String.format("table: %s datacache.partition_duration is %s, nothing need to do",
+                            olapTable.getName(), partitionDuration));
+                    return null;
+                }
+                olapTable.setDataCachePartitionDuration(partitionDuration);
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java:1631]: 1. need write edit log after update.
2. partition duration is also need to update.
`ALTER TABLE table modify partition (p1) set ("xxx"="yyy")`

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

File: fe/fe-core/src/main/java/com/starrocks/persist/OperationType.java [hunk 2]
```diff
@@ -507,6 +507,9 @@ public class OperationType {
     // alter catalog
     public static final short OP_ALTER_CATALOG = 13300;
 
+    // alter datacache info of lake table
+    public static final short OP_ALTER_DATACACHE_PARTITION_DURATION = 13400;
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/persist/OperationType.java:511]: there is no additional processing for data cache partition duration, so i think there is no need to add a new operation type, and use `OP_ALTER_TABLE_PROPERTIES` is fine.

File: fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java [hunk 3]
```diff
@@ -1589,39 +1591,56 @@ public ShowResultSet process(List<AlterClause> alterClauses, Database db, OlapTa
...
-
-            long timeoutSecond = PropertyAnalyzer.analyzeTimeout(properties, Config.alter_table_timeout_second);
-            alterMetaJob = new LakeTableAlterMetaJob(GlobalStateMgr.getCurrentState().getNextId(),
-                    db.getId(),
-                    olapTable.getId(), olapTable.getName(), timeoutSecond,
-                    TTabletMetaType.ENABLE_PERSISTENT_INDEX, enablePersistentIndex);
         } else {
             // shouldn't happen
-            throw new DdlException("only support alter enable_persistent_index in shared_data mode");
+            throw new DdlException("alterClause is not a instance of ModifyTablePropertiesClause");
```
Comment (smartlxh) [inline fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java:1643]: The hit of error maybe not clear, for user don't understand what is `alterClause` and `ModifyTablePropertiesClause` which is internal implemention

**Review by sduzh — DISMISSED** [review]

**Comment by sduzh** [comment]
@Mergifyio rebase

**Comment by sduzh** [comment]
@Mergifyio rebase

**Review by sduzh — DISMISSED** [review]

**Review by wyb — DISMISSED** [review]

**Review by sduzh — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/catalog/OlapTable.java [hunk 4]
```diff
@@ -2993,5 +2993,12 @@ private boolean isEnableFillDataCacheImpl(Partition partition) throws AnalysisEx
         }
         return true;
     }
+
+    public PeriodDuration dataCachePartitionDuration() {
```
[the remark this answers is not in the document]
Comment (stonechen111) [reply fe/fe-core/src/main/java/com/starrocks/catalog/OlapTable.java:2997]: OK

File: fe/fe-core/src/main/java/com/starrocks/catalog/TableProperty.java [hunk 5]
```diff
@@ -776,7 +777,7 @@ public Map<Long, Long> getBinlogAvailaberVersions() {
         return binlogAvailabeVersions;
     }
 
-    public PeriodDuration getDataCachePartitionDuration() {
+    public PeriodDuration dataCachePartitionDuration() {
```
[the remark this answers is not in the document]
Comment (stonechen111) [reply fe/fe-core/src/main/java/com/starrocks/catalog/TableProperty.java:780]: Not necessarily necessary, I will roll it back

**Review by sduzh — DISMISSED** [review]

**Review by kevincai — APPROVED** [review]

**Review by sduzh — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Comment by sduzh** [comment]
@Seaven @kangkaisen PTAL

**Review by Seaven — APPROVED** [review]

**Pull request merged**